src/common/interfaces/endpoint-impl.interface.ts

```diff
@@ -0,0 +1,103 @@
+import { ApiProperty } from '@nestjs/swagger';
+import {
+  IsString,
+  IsNotEmpty,
+  IsEnum,
+  IsOptional,
+  IsBoolean,
+} from 'class-validator';
+import { EndpointHeaderType } from '../enums/endpointHeaderType.enum';
+import { ReqBody, HeaderType, QueryType } from './endpoint.interface';
+
+export class ReqBodyClass implements ReqBody {
+  @IsNotEmpty()
+  @IsString()
+  @ApiProperty()
+  key: string;
+
+  @ApiProperty({
+    oneOf: [
+      { type: 'string' },
+      { type: 'boolean' },
+      { type: 'number' },
+      { type: 'object' },
+      { type: 'symbol' },
+      { type: 'array' },
+    ],
+  })
+  value: string | Date | boolean | number | object | symbol | Array<any>;
+
+  constructor(key: string, value: ReqBody['value']) {
+    this.key = key;
+    this.value = value;
+  }
+}
+
+export class HeaderTypeClass implements HeaderType {
```

Comment:
@devynlab why are you implementing an interface

Reply:
I had to coz of how Swagger interacts with interfaces.., there is no way to document interfaces directly, it leads to some issues at runtime...